tests/unit/testaddonindex.cpp

```diff
@@ -69,7 +69,7 @@ void TestAddonIndex::update() {
 
   QSignalSpy indexUpdatedSpy(&ai, SIGNAL(indexUpdated(QList<AddonData>)));
 
-  ai.update(index, QByteArray());
+  ai.update(index, QByteArray("test signature"));
```

Comment:
Needed to add _something_ here. `QByteArray()` evaluates to `nullptr`.

Reply:
QByteArray() == nullptr is misleading, at least to me because it's not a pointer. Can we use QByteArray.isEmpty() instead?

Reply:
The only reason why it was not misleading to me is because I don't have such a deep knowledge of C++ data structures yet :/ It makes sense now that you explain it, that it is weird... It is the behaviour that I noticed, I could be wrong though. I'll make the changes you suggested and see what happens.